contracts/staking/Staking.sol

```diff
@@ -129,212 +128,165 @@ contract Staking is
         external override
         onlyOwner()
     {
-        require(_bundleRegistry.isRegisteredToken(token, chainId), "ERROR:STK-030:TOKEN_NOT_REGISTERED");
-        require(newStakingRate > 0, "ERROR:STK-031:STAKING_RATE_ZERO");
+        require(_registry.isRegisteredToken(token, chainId), "ERROR:STK-020:TOKEN_NOT_REGISTERED");
+        require(newStakingRate > 0, "ERROR:STK-021:STAKING_RATE_ZERO");
 
         uint256 oldStakingRate = _stakingRate[token][chainId];
         _stakingRate[token][chainId] = newStakingRate;
 
         emit LogStakingStakingRateSet(token, chainId, oldStakingRate, newStakingRate);
     }
 
-    function stakeForBundle(
-        bytes32 instanceId, 
-        uint256 bundleId, 
-        uint256 amount
-    )
+
+    function register(
+        bytes32 targetId, 
+        Target memory target
+    ) 
         external override
     {
-        require(_bundleRegistry.isRegisteredBundle(instanceId, bundleId), "ERROR:STK-040:BUNDLE_NOT_REGISTERED");
-        require(this.isBundleStakingSupported(instanceId, bundleId), "ERROR:STK-041:STAKING_TOO_LATE");
-        require(amount > 0, "ERROR:STK-042:STAKING_AMOUNT_ZERO");
+        require(!this.isTarget(targetId), "ERROR:STK-030:TARGET_ALREADY_REGISTERED");
```

Comment:
I find it confusing that this required check for the target to be 'new' and in error case responds with 'target already registered' and then on line 152 there is a require for the target to be 'registered' (known in registry). 
Maybe `isTargetRegistered` should be renamed to `isKnownInRegistry` and then `isTarget` to `isStakingTarget`. or maybe you have a better idea.

Reply:
i know ... likely need to reimplement registry, staking combination so an object only needs to be registered once ...

Reply:
not within this pr